packages/editor/src/bindings/post-meta.js

```diff
@@ -28,4 +28,49 @@ export default {
 			postId
 		).meta?.[ args.key ];
 	},
+	setValue( { registry, context, args, value } ) {
+		const postId =
+			context?.postId ||
+			registry.select( editorStore ).getCurrentPostId();
+		const postType =
+			context?.postType ||
+			registry.select( editorStore ).getCurrentPostType();
+		registry
+			.dispatch( coreDataStore )
+			.editEntityRecord( 'postType', postType, postId, {
+				meta: {
+					[ args.key ]: value,
+				},
+			} );
+	},
+	lockAttributesEditing( { select, context, args } ) {
```

Comment:
You're right it can be confusing. We could change the intent to just the opposite. Some ideas that come to my mind:
* `allowEditing`
* `canEditSource`
* `canUpdateSource`
* `canUpdateSourceValue`

Any ideas are welcome 🙂

Reply:
One important consideration is that the user might be able to edit source, but not necessarily all posts or individual fields, as permissions are very granular. I don't know if that gets reflected on the client.

I see that permissions are checked only on the post level with `canUserEditEntityRecord `. In effect, `canUserEditValue` or `canEditValue` might resonate better.

Reply:
I've changed the name to `canUserEditValue` in the [latest commit](https://github.com/WordPress/gutenberg/pull/61753/commits/570a4b287de28c6b95a55a1ee0e78781c960054b). It looks better to me. Anyways, the API is still private so we can change that if wanted.

> One important consideration is that the user might be able to edit source, but not necessarily all posts or individual fields, as permissions are very granular. I don't know if that gets reflected on the client.

Do you mean that the current conditions might not be enough? We are assuming that if these conditions are `true`, the user can edit the meta field:
* The user can edit the current post.
* The meta field is publicly exposed in the REST API.

What else should we consider?

Reply:
I was thinking about the [`is_protected_meta`](https://developer.wordpress.org/reference/functions/is_protected_meta/) function that allows filtering, but you are correct that when the individual meta fields gets exposed through REST API for the current user (and their role) then we can safely assume they can edit it 👍🏻